v6 docs updates (#4195)
mikro-orm/mikro-orm: docs/docs/upgrading-v5-to-v6.md

Support for older node versions was dropped. 

> ## ⚠️ TypeScript 4.9+ required

Reviewer: one last thing, let's be progressive, i think i might need some TS 5 features like the `const` generics modifier

```suggestion
## ⚠️ TypeScript 5.0+ required
```

i am also thinking about requiring node 18 right ahead, as 16 will reach eol of half a year (afaik it should have happened much sooner, but postponed). which reminds me that i need to try out v20

console.log(ref.age); // real value is available after flush
```

> ## Changed default PostgreSQL column type for `Date`
> Using a type of `Date` would result in an inferred column type of `timestampz(0)` which is not the default for PostgreSQL and is actually [not recommended](https://wiki.postgresql.org/wiki/Don't_Do_This#Don.27t_use_timestamp.280.29_or_timestamptz.280.29). After upgrading, be prepared for a migration which will change the column type of relevant columns to `timestampz`, which is the default. Should you chose to continue using `timestampz(0)`, simply add a `columnType` to your properties:

Reviewer: ```suggestion
## Changed default PostgreSQL `Date` mapping precision

Previosly, all drivers defaulted the `Date` type mapping to a timestamp with 0 precision (so seconds). This is [discouraged in PostgreSQL](https://wiki.postgresql.org/wiki/Don't_Do_This#Don.27t_use_timestamp.280.29_or_timestamptz.280.29), and is no longer valid - the default mapping without the `length` property being explicitly set is now `timestamptz`, which stores microsecond precision, so equivalent to `timestampz(6)`.

To revert back to the v5 behavior, you can either set the `columnType: 'timestampz(0)'`, or use `length: 0`:
```

console.log(ref.age); // real value is available after flush
```

## Changed default `columnType` for `clock_timestamp`

> If you were previously using `clock_timestamp` as a `defaultRaw` value, you will need to specify `columnType: 'timestampz(0)'` or the next migration creation will convert columns to `timestampz`.

Reviewer: Can you provide some details for this? First time hearing about `clock_timestamp`, but it sounds weird that a `defaultRaw` would affect column type anyhow. What changed in v6 is the runtime default (property initializer) is used to infer the database default as well as column type. But only if you don't provide your own options, and `defaultRaw` in particular shouldn't affect this at all.

> you will need to specify `columnType: 'timestampz(0)'` or the next migration creation will convert columns to `timestampz`.

This has nothing to do with the defaults you set, and indeed, this changed - postgres no longer defaults to 0 length for datetimes. #3832

Author: It's something my buddy introduced me to, it basically takes the time on the db server when you insert the record... which I'm just now realizing is a Postgres specific thing.

Here was what my entity used to have:
```ts
export abstract class Node<T extends AnyEntity> extends BaseEntity<T, 'id'> {
// ...
@Property({ defaultRaw: 'clock_timestamp()' })
public createdAt!: Date;
// ...
}
```

On 5.x I was up to date and running `mikro-orm migration:create` resulted in no changes. After the upgrade (and without changing anything) the same command triggered a migration which targeted all the tables based on entities that extend my `Node` class. The migration wanted to change every column from `timestampz(0)` to `timestampz`. Setting the `columnType` explicitly resulted in a no-op when I ran the migration command again.

Maybe the issue is what `Date` translates to from a `columnType` perspective? Like whatever you're using to do that changed?

Reviewer: There is no issue, you dont have to explain it more :) Have you checked the PR I linked? The default datetime length in postgres changed, thats all.

Author: Ah ok cool, and nope I missed the PR. Just looked; want me to update the above to reflect what was changed in that PR so there's *some* mention of it?

Reviewer: Yeah should be mentioned in there for sure, it just needs to be correct - it is not about `default` or `defaultRaw` options, it is about the default length in postgres, which is now aligned with how postgres behaves instead of how other drivers behave. 

I guess I will add some hooks to the discovery so this can be easily overridden globally.

Author: Honestly until you get feedback, I wouldn't bother... I didn't want the extra migration, but all it does it replace a couple of zeros with other numbers; the structure should be exactly the same. Honestly not even sure how/why that differentiation exists 😬 I'll re-word and feel free to tweak it after if I misused a term or something

Reviewer: https://wiki.postgresql.org/wiki/Don't_Do_This#Don.27t_use_timestamp.280.29_or_timestamptz.280.29

Author: Yea I saw that from the PR you mentioned; until someone asks for that explicitly, I personally wouldn't bother haha I'm actually going to revert my change in my project and migrate to the correct column type.

My changes have been pushed; take a look and let me know what you think

Author: @B4nan are you ready to resolve this thread?

console.log(ref.age); // real value is available after flush
```

## Changed default PostgreSQL column type for `Date`
Using a type of `Date` would result in an inferred column type of `timestampz(0)` which is not the default for PostgreSQL and is actually [not recommended](https://wiki.postgresql.org/wiki/Don't_Do_This#Don.27t_use_timestamp.280.29_or_timestamptz.280.29). After upgrading, be prepared for a migration which will change the column type of relevant columns to `timestampz`, which is the default. Should you chose to continue using `timestampz(0)`, simply add a `columnType` to your properties:

```ts

> @Property({ columnType: 'timestamptz(0)' })

Reviewer: maybe we should suggest this instead?
```suggestion
@Property({ length: 0 })
```

Author: If that does the same thing, absolutely; I just wasn't aware that had an impact. I'll try it locally and make that change if it works as expected
